### ISSUE TITLE
MonadCatchIO-transformers dependency and GHC v7.7

### ISSUE THREAD

**Issue by ggreif** [ekmett/lens#301]
Looks like the author of MonadCatchIO-transformers is about to deprecate the package.

Background:

base 4.7 will remove the deprecated `block`
and `unblock` functions, so there will be a failure:

```
Resolving dependencies...
Configuring MonadCatchIO-transformers-0.3.0.0...
Building MonadCatchIO-transformers-0.3.0.0...
Preprocessing library MonadCatchIO-transformers-0.3.0.0...
[1 of 1] Compiling Control.Monad.CatchIO (
src/Control/Monad/CatchIO.hs, dist/build/Control/Monad/CatchIO.o )

src/Control/Monad/CatchIO.hs:54:13: Not in scope: `E.block'

src/Control/Monad/CatchIO.hs:55:13: Not in scope: `E.unblock'
Failed to install MonadCatchIO-transformers-0.3.0.0
cabal: Error: some packages failed to install:
MonadCatchIO-transformers-0.3.0.0 failed during the building phase. The
exception was:
ExitFailure 1
```

I guess it is time to start thinking about a replacement.
